VerilogCodingStyle.md

```diff
@@ -1390,10 +1392,14 @@ Use the Verilog-2001 combined port and I/O declaration style. Do not use the
 Verilog-95 list style. The port declaration in the module statement should fully
 declare the port name, type, and direction.
 
-The opening parenthesis should be on the same line as the module declaration,
-and the first port should be declared on the following line.
+The opening parenthesis may be on the same line as the module declaration,
```

Comment:
I'm with @msfschaffner on this one that it needs more discussion. We
can spawn to a different PR or leave this one open for a while to get other
opinions. @asb @imphil @eunchan @tjaychen @GregAC @martin-lueker 

I'm not an emacs user, so please weigh in if this is a true hardship. Even if
it is hard to avoid, I don't believe it should be in our rationale explicitly as
on the line below.

Reply:
please include me if this spawns into another issue
@sjgitty  with emacs default is to align following lines to the opening parenthese
so an example could be

```
module this_is_my_mode (
                        input logic a,
                        input logic b,
                        output logic c
                         );
```

vs

```
module this_is_my_mode
   (
    input logic a,
    input logic b,
    output logic c
   ); 

```
this can of cause be edited in the mode file - but then one would have to change this every time going to or from opentitan work.

Reply:
@rswarbrick as you're a heavy Emacs user: can we provide a project-specific config for OpenTitan to make it easier for Emacs user to get the right style from verilog-mode?

Reply:
As far as I can tell, this isn't trivial to do in [verilog-mode](https://github.com/veripool/verilog-mode) (the major mode that everyone uses for editing Verilog in Emacs). When calculating how to do indentation, it currently decides that the ports are in a syntactic context that it calls `cparenexp` (parenthesized expression context), and always lines up with the opening paren. However, it's maintained by Wilson Snyder (of Verilator fame), who's super helpful if you send him a pull request.

This has been annoying me too, so I intend to get around to it soonish (hah..). I'd be thrilled if someone else did so first! :-) In the meantime, rectangular text commands and judicious use of  `indent-rigidly` (C-x TAB) seems to be a reasonable workaround.

I'm not sure what Rasmus means by "mode file". If we have a working way to setup verilog mode in the "lowRISC style", the usual thing would be to write a function like this:

    (defun setup-lowrisc-verilog-style ()
      (when (lowrisc-repo-p (buffer-file-name))
        (setq-local my-variable 'some-value)))

and install it on `verilog-mode-hook`. This runs when you open a Verilog file (well, when you enable verilog-mode, but that should be equivalent) and sets up "lowRISC mode" if needed.

Obviously, you also have to write an implementation of `lowrisc-repo-p`. I'd probably do something like walk upwards through the path looking for the first directory with a `.git` subdirectory and then checking whether that directory name was in a specified list (`ibex`, `opentitan`, ...). Alternatively, we could use a dir-locals file checked in to the repo which set an "I'm a lowRISC repository" flag.

If there are significantly more people that would use this than me and Rasmus, we could even have a simple "emacs setup" repository somewhere.